### ISSUE TITLE
The request must contain the parameter LaunchPermission

### ISSUE THREAD

**Issue by gdbtek** [mitchellh/packer#2305]
I have been using packer **0.7.5** to build AMI weekly and after I upgraded to **0.8.0**, I got following errors. Any idea what seems to be the issue?

```
==> amazon-ebs: Stopping the source instance...
==> amazon-ebs: Waiting for the instance to stop...
==> amazon-ebs: Enabling Enhanced Networking...
==> amazon-ebs: Creating the AMI: CCMUI-OPS-CCM-WEB-1435134775
    amazon-ebs: AMI: ami-9f4bb3f4
==> amazon-ebs: Waiting for AMI to become ready...
==> amazon-ebs: Copying AMI (ami-9f4bb3f4) to other regions...
    amazon-ebs: Copying to: ap-northeast-1
    amazon-ebs: Copying to: eu-west-1
    amazon-ebs: Waiting for all copies to complete...
==> amazon-ebs: Modifying attributes on AMI (ami-9f4bb3f4)...
    amazon-ebs: Modifying: description
    amazon-ebs: Modifying: users
==> amazon-ebs: Error modify AMI attributes: MissingParameter: The request must contain the parameter LaunchPermission
==> amazon-ebs: 	status code: 400, request id: []
==> amazon-ebs: Deregistering the AMI because cancelation or error...
==> amazon-ebs: Terminating the source AWS instance...
==> amazon-ebs: Cleaning up any extra volumes...
==> amazon-ebs: Deleting temporary keypair...
Build 'amazon-ebs' errored: Error modify AMI attributes: MissingParameter: The request must contain the parameter LaunchPermission
	status code: 400, request id: []

==> Some builds didn't complete successfully and had errors:
--> amazon-ebs: Error modify AMI attributes: MissingParameter: The request must contain the parameter LaunchPermission
	status code: 400, request id: []

==> Builds finished but no artifacts were created.

```

**Comment by asagage** [comment]
I'm getting this too.  Seems like it has to do with assigning the ami_users to the ami.

**Comment by catsby** [comment]
Thanks for reporting this, I'm taking a look now

**Comment by asagage** [comment]
I think this commit alters expansion of arrays in variables, not sure if this is related or not. The ami_users for instance is an array of strings.   https://github.com/mitchellh/packer/commit/05c478c8f797297727d5fd76d63eed50f39aecba

**Comment by catsby** [comment]
I have a fix for this, I'm writing an acceptance test now, should be available soon

**Comment by catsby** [comment]
This should be fixed with https://github.com/mitchellh/packer/pull/2308, if anyone can take that for a spin I'd greatly appreciate it!

**Comment by catsby** [comment]
Went ahead and merged #2308, let me know if you're still hitting this. Thanks!

**Comment by gdbtek** [comment]
@catsby are we going to have a new patched version of 0.8.0 for this fix?

**Comment by catsby** [comment]
We'll do a 0.8.1 release with bug fixes like this in the near future, I suspect next week sometime
